src/main/java/ch/njol/skript/util/PotionEffectUtils.java

```diff
@@ -99,32 +95,30 @@ public static PotionEffectType parseByEffectType(PotionEffectType t) {
 		return null;
 	}
 	
-	@SuppressWarnings("null")
-	public static String toString(final PotionEffectType t) {
+	public static String toString(PotionEffectType t) {
 		return names[t.getId()];
 	}
 	
 	// REMIND flags?
-	@SuppressWarnings("null")
-	public static String toString(final PotionEffectType t, final int flags) {
+	public static String toString(PotionEffectType t, int flags) {
 		return names[t.getId()];
 	}
-	
+
 	public static String toString(PotionEffect potionEffect) {
 		StringBuilder builder = new StringBuilder();
 		if (potionEffect.isAmbient())
 			builder.append("ambient ");
 		builder.append("potion effect of ");
 		builder.append(toString(potionEffect.getType()));
-		
 		builder.append(" of tier ").append(potionEffect.getAmplifier() + 1);
-		
 		if (!potionEffect.hasParticles())
 			builder.append(" without particles");
-		builder.append(" for ").append(Timespan.fromTicks_i(potionEffect.getDuration()));
+		builder.append(" for ").append(potionEffect.isInfinite() ? "infinity" : Timespan.fromTicks_i(Math.abs(potionEffect.getDuration())));
```

Comment:
It's not safe to call isInfinite without a version check right? Also, I'm pretty sure it's just a wrapper for checking `getDuration() == PotionEffect.INFINITE_DURATION`, so you could just do that instead